webapp/src/Config/Resource/OptionalFileResource.php

```diff
@@ -0,0 +1,45 @@
+<?php declare(strict_types=1);
+
+namespace App\Config\Resource;
+
+use Symfony\Component\Config\Resource\FileExistenceResource;
+use Symfony\Component\Config\Resource\FileResource;
+use Symfony\Component\Config\Resource\SelfCheckingResourceInterface;
+
+/**
+ * This resource type is based on Symfony's built in FileResource but also
+ * tracks presence of the file
+ *
+ * @see FileResource
+ * @see FileExistenceResource
+ */
+class OptionalFileResource implements SelfCheckingResourceInterface
+{
+    private $resource;
+
+    private $exists;
+
+    public function __construct(string $resource)
+    {
+        $this->resource = $resource;
+        $this->exists = file_exists($resource);
+    }
+
+    public function __toString()
+    {
+        return $this->resource;
+    }
+
+    public function getResource()
+    {
+        return $this->resource;
+    }
+
+    public function isFresh($timestamp)
+    {
+        if (file_exists($this->resource) !== $this->exists) {
+            return false;
+        }
+        return false !== ($filemtime = @filemtime($this->resource)) && $filemtime <= $timestamp;
```

Comment:
The `false !== ` seems superfluous. Maybe better to write this out more, since I don't really get what this method is trying to do.

Reply:
Sorry, was a copy paste from the file I linked. But I agree it’s better to write it out

Reply:
Or... we can accept that you clear the cache when changing docs.yml? Just like you need to do when adding a banner image. So we also prevent checking whether the page is modified on every page load?

Reply:
Adding a banner does *not* require clearing the cache. The main reason we add this as a parameter is that we do not have to decode yaml on every (team) page, which is kinda costly.
I can remove the resource check itself but I think we do need to set this as a parameter to not decrease performance

Reply:
Our docs claim that you need to do that.

Reply:
Then I think the docs are wrong. Although come to think of it: I think this is true in production. In development you don’t have to.
I am fine with removing the check, but then we should remove it for custom css/js and the banner as well to make it consistent (and accept you also need to clear cache on dev mode).
Opinions?

Reply:
I'm not entirely sure I follow this. This is server-side caching, right? So nothing would change client-side? And what would you need to do server side to clear the cache? Also, if this is cached, where is it stored then, and how much is the performance impact?

Reply:
This is the symfony cache. It’s stored in webapp/var/cache, just like for example the routes. Impact I don’t know exactly, but parsing files is never free.
As said I’m totally fine with removing the resource check. It would still be cached but then not automatically updated in dev mode if the file changed. I can also remove the whole cache, but then we should be consistent and also do it for the custom
CSS / JavaScript and banner, which also has a performance impact: globs and file existence check on each page. Again, not sure how much the impact is, but I also wonder why we should NOT cache it? After I update the code in this file to make it more readable